### DISCUSSION TITLE
How to invalidate OIDC scopes after giving a user new ones?

### DISCUSSION THREAD

**Discussion by dblsaiko** [kanidm/kanidm#2807] [Q&A] [answered]
I've just set up OIDC with Nextcloud as a test, first with the basic scopes - email, openid, profile - then added a nextcloud-admins group which gives the admin scope and added my user to that. However, it still continues using the old scopes that I logged in with before, without the admin one:

```
May 30 18:25:24 vineta kanidmd[864478]: 36a117cc-3234-4f65-8a14-fb183f7579c3 INFO        │  │  ┕━ ｉ [info]: A valid limited session value exists for this token | event_tag_id: 10
May 30 18:25:24 vineta kanidmd[864478]: 36a117cc-3234-4f65-8a14-fb183f7579c3 INFO        │  ┕━ ｉ [info]: User has previously consented, permitting with scopes: email,openid,profile | event_tag_id: 3
```

How do you make it recognize that the user now has a new scope?

**Comment by dblsaiko** [comment, marked as answer]
Ah, I think I'm doing it wrong, I just saw that there's more configurable options in the Nextcloud plugin for this kind of stuff.

EDIT: Yup. I had to do this instead:

```
kanidm system oauth2 update-claim-map nextcloud groups nextcloud-admins admin
kanidm system oauth2 update-claim-map-join nextcloud groups array
```

and then add "groups" to "Extra claims" in the nextcloud plugin settings and enable group provisioning.

**Reply by Firstyear** [reply]
Yeah, I think you have solved this but you'd likely need to logout and then backin here to force the access / refresh token pairs to be re-issued with the new scopes.

**Reply by dblsaiko** [reply]
You mean log out from Kanidm? I didn't try that, but I tried logging out of Nextcloud multiple times.

**Reply by Firstyear** [reply]
No, I meant logging out of nextcloud :)

**Reply by dblsaiko** [reply]
Hm, yeah in that case that didn't work. That's what the log snippet is from in my original post in fact, log-in specifically (there's nothing in the kanidm log when logging out of Nextcloud, should there be?)

**Reply by Firstyear** [reply]
That will be up to nextcloud - it should be binding the session to the access token and periodically validating the access and refresh tokens. And when you logout of nextcloud, it should be requesting a new access/refresh pair from kanidm. So that sounds more like a nextcloud issue?
